cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala

```diff
@@ -126,23 +135,59 @@ class CallSubqueryTest extends DocumentingTest {
-          #This example query creates five `Clone` nodes for each existing person.
-          #The aggregation ensures that cardinality is not changed by the subquery.
-          #Without this, the result would be five times as many rows.""".stripMargin('#'))
+    section("Cardinality and aggregation", "subquery-aggregation") {
+      p(
+        """Returning subqueries change the cardinality of the query: The results of each subquery call are collected and together they give the result of the
+          |enclosing query.""".stripMargin)
+      p("""The following example lists each connected person that each person has:""")
       query("""MATCH (p:Person)
               #CALL {
-              #  UNWIND range(1, 5) AS i
-              #  CREATE (c:Clone)
-              #  RETURN count(c) AS numberOfClones
+              #  WITH p
+              #  MATCH (p)-[:FRIEND_OF]-(c:Person)
+              #  RETURN c.name AS friend
               #}
-              #RETURN p.name, numberOfClones""".stripMargin('#'),
+              #RETURN p.name, friend""".stripMargin('#'),
       ResultAssertions(r => r.toSet should equal(Set(
-          Map("p.name" -> "Alice", "numberOfClones" -> 5),
-          Map("p.name" -> "Bob", "numberOfClones" -> 5),
-          Map("p.name" -> "Charlie", "numberOfClones" -> 5),
-          Map("p.name" -> "Dora", "numberOfClones" -> 5)
+          Map("p.name" -> "Alice", "friend" -> "Bob"),
+          Map("p.name" -> "Bob", "friend" -> "Alice"),
+        )))) { resultTable() }
+      p(
+        """The number of results of the subquery changed the number of results of the enclosing query: Instead of 4 rows, one for each node), there are
+          |now 2 rows which were found for Alice and Bob respectively. The nodes for Charlie and Dora are not returned.""".stripMargin)
+      p(
+        """Besides changing the set of matched nodes, which we can also express using a simple query, we can also use the subquery to aggregate on each of
+          |the matches of the enclosing query in isolation. As we get one result for each call of the subquery, the cardinality does not differ from the
+          |enclosing query's cardinality:""".stripMargin)
+      query("""MATCH (p:Person)
+              #CALL {
+              #  WITH p
+              #  MATCH (p)--(c)
+              #  RETURN count(c) AS numberOfConnections
+              #}
+              #RETURN p.name, numberOfConnections""".stripMargin('#'),
+        ResultAssertions(r => r.toSet should equal(Set(
+          Map("p.name" -> "Alice", "numberOfConnections" -> 2),
+          Map("p.name" -> "Bob", "numberOfConnections" -> 1),
+          Map("p.name" -> "Charlie", "numberOfConnections" -> 1),
+          Map("p.name" -> "Dora", "numberOfConnections" -> 0)
+        )))) { resultTable() }
+    }
+
+    section("Unit subqueries and side-effects", "subquery-unit") {
+      p("""Unit subqueries do not return any value and are therefore used for their side effects.""")
```

Comment:
Perhaps we should use "rows" as the thing that is returned
```suggestion
      p("""Unit subqueries do not return any rows and are therefore used for their side effects.""")
```